src/timeline/timeline.tsx

```diff
@@ -68,21 +71,26 @@ export const Timeline = ({
             );
         });
     };
+
     const renderItems = () =>
         items.map((item: TimelineItemProps, index) => {
-            const { title, content, statuses } = item;
+            const { title, content, statuses, variant: _variant } = item;
             const circleIndicatorTestId = baseIndicatorTestId
                 ? `circleindicator${index + 1}_div_${baseIndicatorTestId}`
                 : "circleindicator";
+            const variant =
+                _variant || (index === 0 ? "current" : "upcoming-active");
 
             return (
                 <TimelineItem key={`timeline-item-${index}`}>
                     <TimelineIndicators>
                         <CircleIndicator
                             data-testid={circleIndicatorTestId}
-                            $filled={index === 0}
-                        />
-                        <LineIndicator />
+                            $variant={variant}
+                        >
+                            <IconTick />
```

Comment:
instead of a fixed icon, can have a `renderIcon(variant)` with a switch case that returns `TickIcon` for completed, otherwise default null

in `timeline.style.tsx`, you can simply style all svg children with color, width and height without having to toggle the display

Reply:
i see. No needed to render into DOM if didn't use it! thanks!